patches/api/0416-Brigadier-based-command-API.patch

```diff
@@ -0,0 +1,758 @@
+      * @return the default message
+      */
+     @NotNull
++    @Deprecated // Used only for bukkit commands
+     public static net.kyori.adventure.text.Component permissionMessage() {
+         return server.permissionMessage();
+     }
+@@ -2454,6 +2455,11 @@ public final class Bukkit {
+     public static @NotNull org.bukkit.potion.PotionBrewer getPotionBrewer() {
+         return server.getPotionBrewer();
+     }
++
++    @NotNull
++    public static com.mojang.brigadier.CommandDispatcher<io.papermc.paper.command.brigadier.CommandSourceStack> getCommandDispatcher() {
++        return server.getCommandDispatcher();
++    }
+     // Paper end
+ 
+     @NotNull
+diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
+index 9f7029d327b55c8b1b9cfef8ea2678b87dad8ddb..f6062473ad743083849a5d51d1299ee5e24b64c5 100644
+--- a/src/main/java/org/bukkit/Server.java
++++ b/src/main/java/org/bukkit/Server.java
+@@ -2029,6 +2029,7 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
+      *
+      * @return the default message
+      */
++    @Deprecated // Paper - only used for bukkit commands
+     @NotNull net.kyori.adventure.text.Component permissionMessage();
+ 
+     /**
+@@ -2134,5 +2135,8 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
+      * @return the potion brewer
+      */
+     @NotNull org.bukkit.potion.PotionBrewer getPotionBrewer();
++
++    @NotNull
++    com.mojang.brigadier.CommandDispatcher<io.papermc.paper.command.brigadier.CommandSourceStack> getCommandDispatcher();
+     // Paper end
+ }
+diff --git a/src/main/java/org/bukkit/UnsafeValues.java b/src/main/java/org/bukkit/UnsafeValues.java
+index b92255a9c87620f46adb140689b1cd328a476d61..50807b2718676335d981aaca6697718fce49781f 100644
+--- a/src/main/java/org/bukkit/UnsafeValues.java
++++ b/src/main/java/org/bukkit/UnsafeValues.java
+@@ -244,5 +244,7 @@ public interface UnsafeValues {
+      * @throws IllegalStateException if no biome by the given key is registered.
+      */
+     void setBiomeKey(RegionAccessor accessor, int x, int y, int z, NamespacedKey biomeKey);
++
++    java.util.Map<String, org.bukkit.command.Command> getCommandMirrorMap();
+     // Paper end
+ }
+diff --git a/src/main/java/org/bukkit/command/Command.java b/src/main/java/org/bukkit/command/Command.java
+index b791358f90fe92bc2264d9a26492245763813af3..28fc568a033f0c8c58292983b5b5f2adec1987c9 100644
+--- a/src/main/java/org/bukkit/command/Command.java
++++ b/src/main/java/org/bukkit/command/Command.java
+@@ -488,4 +488,12 @@ public abstract class Command {
+     public String toString() {
+         return getClass().getName() + '(' + name + ')';
+     }
++
++    // Paper start
++    @org.jetbrains.annotations.ApiStatus.Internal
++    public boolean canBeOverriden() { return false; }
++
++    @org.jetbrains.annotations.ApiStatus.Internal
++    public String registeredNamespace;
```

Comment:
public field in the API sounds like a bad idea, this should somehow be refactored out or hidden. I haven't gotten far enough to know its full purpose yet.

Reply:
Removed